api/src/main/java/org/jboss/pnc/api/deliverablesanalyzer/dto/AnalysisReport.java

```diff
@@ -46,25 +47,31 @@ public class AnalysisReport implements Serializable {
     /** Results of the analysis (if analysis was successful) */
     private List<@Valid FinderResult> results;
 
-    /** Flag indicating if analysis was finished successfully */
-    private boolean success;
+    /** Enum indicating the resulting analysis status */
+    private ResultStatus resultStatus;
 
     /** Exception log if any caught exception occurs during the process */
     private ExceptionResolution exceptionResolution;
 
     public AnalysisReport(List<FinderResult> results) {
         this.results = results;
-        success = true;
+        resultStatus = ResultStatus.SUCCESS;
     }
 
     public AnalysisReport() {
-        success = false;
+        resultStatus = ResultStatus.FAILED;
     }
 
-    public static AnalysisReport failWithResolution(ExceptionResolution exceptionResolution) {
+    public static AnalysisReport processWithResolution(
+            ResultStatus resultStatus,
+            ExceptionResolution exceptionResolution) {
```

Comment:
exceptionResolution is only for failed statuses, this method should not allow successful statuses